app.js

```diff
@@ -123,13 +244,153 @@ var budgetController = (function () {
+			}
+
+			newHtml = html.replace("%id%", obj.id);																					// Replace placeholder with data
+			newHtml = newHtml.replace("%description%", obj.description);
+			newHtml = newHtml.replace("%value%", formatNumber(obj.value, type));
+
+			document.querySelector(element).insertAdjacentHTML("beforeend", newHtml);				// Insert HTML into the DOM
+
+		},
+
+		deleteListItem: function (selectorID) {
+			var element = document.getElementById(selectorID);
+			element.parentNode.removeChild(element);
+		},
+
+		clearFields: function () {
+			var fields, fieldsArr;
+
+			fields = document.querySelectorAll(DOMstrings.inputDescription + ", " + DOMstrings.inputValue);			// "querySelectorAll" returns a node list, not and array
+			fieldsArr = Array.prototype.slice.call(fields);																											// Make "slice" think that this is an array and not a list
+			fieldsArr.forEach(function (current, index, array) {																									// Clear the fields
+				current.value = "";
+			});
+			fieldsArr[0].focus();																																								// Focus goes back to the first element ==> item description
+		},
+
+		displayBudget: function (obj) {
+			var type;
+			obj.budget > 0 ? type = "inc" : type = "exp";
+			document.querySelector(DOMstrings.budgetLabel).textContent = formatNumber(obj.budget, type);
+			document.querySelector(DOMstrings.incomeLabel).textContent = formatNumber(obj.totalInc, "inc");
+			document.querySelector(DOMstrings.expensesLabel).textContent = formatNumber(obj.totalExp, "exp");
+			if (obj.percentage > 0) {
+				document.querySelector(DOMstrings.percentageLabel).textContent = obj.percentage + "%";
+			} else {
+				document.querySelector(DOMstrings.percentageLabel).textContent = "---";
+			}
+		},
+
+		displayPercentages: function (percentages) {
+			var fields = document.querySelectorAll(DOMstrings.expensesPercLabel);
+			nodeListForEach(fields, function (current, index) {
+				if (percentages[index] > 0) {
+					current.textContent = percentages[index] + "%";
+				} else {
+					current.textContent = "---";
+				}
+			});
+		},
+
+		displayDate: function () {
+			var now, month, months, year;
+			now = new Date();
+			month = now.getMonth();
+			months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+			year = now.getFullYear();
+			document.querySelector(DOMstrings.dateLabel).textContent = months[month] + " " + year;
+		},
+
+		changedType: function () {
+			var fields = document.querySelectorAll(
+				DOMstrings.inputType + "," +
+				DOMstrings.inputDescription + "," +
+				DOMstrings.inputValue
+			);
+
+			nodeListForEach(fields, function (current) {
+				current.classList.toggle("red-focus");
+			});
+
+			document.querySelector(DOMstrings.inputButton).classList.toggle("red");
+		},
+
+		getDOMstrings: function () {
+			return DOMstrings;
+		}
+	};
+
+///
 //don`t use var, use let or const instead

```

Comment:
// in this case variable with vat can be called globally. That`s why use const or let